website/integrations: add Immich (#7829)
goauthentik/authentik: website/integrations/services/immich/index.md

---
title: Immich
---

<span class="badge badge--secondary">Support level: Community</span>

## What is Immich

> > Immich is a self-hosted backup solution for photos and videos on mobile devices

Author: Fixed in [07febfa](https://github.com/goauthentik/authentik/pull/7829/commits/07febfaaac6b7c4dade7d39178006215ebf728e3)

>
> -- https://immich.app/

## Preparation

The following placeholders will be used:

-   `https://immich.company` is the URL used to access the Immich instance.
-   `authentik.company` is the FQDN of the authentik install.

> ## authentik Configuration

Reviewer: ```suggestion
## authentik configuration
```

Reviewer: Sorry I know it was capitalized in the template; I have fixed that now. :-)

Author: Fixed in [07febfa](https://github.com/goauthentik/authentik/pull/7829/commits/07febfaaac6b7c4dade7d39178006215ebf728e3)

> 1. Create a new OAuth2/OpenID Provider

Reviewer: ```suggestion
1. Create a new OAuth2/OpenID Provider using the following settings:
```

Author: Fixed in [07febfa](https://github.com/goauthentik/authentik/pull/7829/commits/07febfaaac6b7c4dade7d39178006215ebf728e3)

    - **Name**: Immich
    - **Authentication flow**: default-authentication-flow
    - **Authorization flow**: default-provider-authorization-explicit-consent
    - **Client type**: Confidential
    - **Client ID**: Either create your own Client ID or make a note of the auto-populated one
    - **Client Secret**: Either create your own Client Secret or make a note of the auto-populated one
    - **Redirect URIs/Origins (RegEx)**:

>       _Please note that the following URIs are just examples. Ensure to include all of the domains / URLs you will be using to access Immich._

Reviewer: ```suggestion
      _Please note that the following URIs are just examples. Be sure to include all of the domains / URLs that you will use to access Immich._
```

Author: Fixed in [07febfa](https://github.com/goauthentik/authentik/pull/7829/commits/07febfaaac6b7c4dade7d39178006215ebf728e3)

        - app.immich:/
        - https://immich.company/auth/login
        - https://immich.company/user-settings
    - **Signing Key**: authentik Self-signed Certificate
    - Leave everything else as default

> 2. Open the new provider you've just created

Reviewer: ```suggestion
2. Open the new provider you've just created.
```

Author: Fixed in [07febfa](https://github.com/goauthentik/authentik/pull/7829/commits/07febfaaac6b7c4dade7d39178006215ebf728e3)

> 3. Make a note of the **OpenID Configuration Issuer**

Reviewer: ```suggestion
3. Make a note of the **OpenID Configuration Issuer**.
```

Author: Fixed in [07febfa](https://github.com/goauthentik/authentik/pull/7829/commits/07febfaaac6b7c4dade7d39178006215ebf728e3)

> ## Service Configuration

Reviewer: ```suggestion
## Immich configuration
```

Immich documentation can be found here: https://immich.app/docs/administration/oauth

> 1. Go to Administration > Settings > OAuth Authentication

Reviewer: ```suggestion
1. In Immich, navigate to **Administration** > **Settings** > **OAuth Authentication**.
```

Author: Fixed in [07febfa](https://github.com/goauthentik/authentik/pull/7829/commits/07febfaaac6b7c4dade7d39178006215ebf728e3)

2. Configure Immich as follows:
    - **Issuer URL**: Populate this field with the `OpenID Configuration Issuer`

>     - **Client ID**: Enter yout Client ID from authentik here

Reviewer: ```suggestion
    - **Client ID**: Enter your Client ID from authentik
```

Author: Fixed in [07febfa](https://github.com/goauthentik/authentik/pull/7829/commits/07febfaaac6b7c4dade7d39178006215ebf728e3)

>     - **Client Secret**: Enter yout Client Secret from authentik here

Reviewer: ```suggestion
    - **Client Secret**: Enter your Client Secret from authentik
```

Author: Fixed in [07febfa](https://github.com/goauthentik/authentik/pull/7829/commits/07febfaaac6b7c4dade7d39178006215ebf728e3)

---
title: Immich
---

<span class="badge badge--secondary">Support level: Community</span>

## What is Immich

> Immich is a self-hosted backup solution for photos and videos on mobile devices
>
> -- https://immich.app/

## Preparation

The following placeholders will be used:

-   `immich.company` is the FQDN of the Immich install.

> -   `192.168.0.2:2283` is the local IP address & port of the Immich install.

Reviewer 2: ```suggestion
```

Let's remove any custom environment examples, and only stick to using `immich.company`.

Author: I wanted to include the local IP in some way as this needs to be added to the redirect URIs to allow for local access using the IP address rather than the FQDN. Is there a way we can compromise here? Perhaps a placeholder just in the URI redirect section to say "add your local IP here too"?

Author: That's fair enough - Double checking the Immich documentation, it does just show the local IP as an example along with `localhost` too
```
Redirect URIs should contain all the domains you will be using to access Immich. Some examples include
```
So I'll try incorporate this wording into the URI section here too.

-   `authentik.company` is the FQDN of the authentik install.

## authentik Configuration

1. Create a new OAuth2/OpenID Provider
    - **Name**: Immich
	- **Authentication flow**: default-authentication-flow
	- **Authorization flow**: default-provider-authorization-explicit-consent
	- **Client type**: Conidential
	- **Client ID**: Either create your own Client ID or make a note of the auto-populated one
	- **Client Secret**: Either create your own Client Secret or make a note of the auto-populated one
	- **Redirect URIs/Origins (RegEx)**:
	  - app.immich:/

> 	  - http://192.168.0.2:2283/auth/login
> 	  - http://192.168.0.2:2283/user-settings

Reviewer 2: ```suggestion
```

Same as above.
